src/watcher/watcher.ts

```diff
@@ -146,8 +151,8 @@ export function createWatcher(ctx: Context) {
       const userOptions = props.chokidarOptions || {};
       const finalOptions = { ...defaultOpts, ...userOptions };
 
-      const root = props.root || [`${env.APP_ROOT}`];
-      const watcher = chokidarWatch(root, finalOptions);
+      const watchRoot = absRoots || `${env.APP_ROOT}`;
```

Comment:
shouldn't absRoots be defined before this line? The default config should be
`env.SCRIPT_PATH`

Reply:
I didn't change the default, it was already `env.APP_ROOT` so I left it.  I can change it if you think I should, but when I thought about it, `APP_ROOT` made sense as it is probably more likely to do what the user wants since in almost all cases, `SCRIPT_PATH` is inside `APP_ROOT` anyway, and if the user wants fine-grained control, they can mess with `include`/`ignore`

`absRoot` can be `undefined` here because `root` might be `undefined` because `props.root` might be `undefined`.

And so if the user doesn't specify a root, this is where the defaulting ends up happening.

Reply:
sorry, I mean `watchRoot` should always contain APP_ROOT regardless of user settings. The watcher detects changes to `fuse.ts`. So we should always have it there.

Reply:
Gotcha, I'll get that

Reply:
This became a little more complicated.

Because `SCRIPT_PATH` might be the same as `APP_ROOT` and so you can easily end up with:

```
[
  `/my/app/path`,
  `/my/app/path`, // <-- redundant
]
```

But removing duplicates isn't really enough because one root might contain the other.  I.e. `SCRIPT_PATH` might be _inside_ `APP_ROOT`

```
[
  `/my/app/path`,
  `/my/app/path/scripts` // <-- also redundant
]
```
So the best solution is to remove all _redundant_ paths.  I added a utility function, `excludeRedundantFolders()` to do that and a test to check that it works.